crates/doc/src/ptr.rs

```diff
@@ -502,7 +506,7 @@ mod test {
         }
 
```

Comment:
add back the `/foo/-` case here by manually building up the tokens? You can't implicitly build them by parsing a pointer as it used to do, but you can still build it manually.

Also add a case covering creation with NextProperty. It'd return None, right?

Reply:
Got totally side-tracked and forgot to finished this. What do you mean by adding `/foo/-` back? That pointer doesn't make any sense. `/foo` is an array, and `-` is interpreted as an object field called `-`, which you can't use to index an array

Reply:
I meant build it up from tokens, rather than parsing, so that it's the NextIndex variant.

It's possible to build such a pointer and pass it to create, right? Let's cover that in the test.